modular_skyrat/modules/modular_implants/code/nifsofts/huds.dm

```diff
@@ -0,0 +1,131 @@
+/datum/nifsoft/hud
+	name = "blank HUD"
+	program_desc = "Provides the overlay of a hud on compatible eyewear."
+	compatible_nifs = list(/obj/item/organ/internal/cyberimp/brain/nif/standard)
+	active_mode = TRUE
+	active_cost = 0.5
+	ui_icon = "eye"
+	/// Do we need to check if the user is wearing compatible eyewear?
+	var/eyewear_check = TRUE
+	/// What kind of HUD are we adding when the NIFSoft is activated?
+	var/hud_type
+	/// What is the HUD trait we are adding when the NIFSoft is activated?
+	var/hud_trait
+	/// A list of traits that we want to add while the NIFSoft is active. This is here to emulate things like sci-goggles
+	var/list/added_eyewear_traits = list()
+
+/datum/nifsoft/hud/activate()
+	var/obj/item/clothing/glasses/worn_glasses = linked_mob.get_item_by_slot(ITEM_SLOT_EYES)
+	if(eyewear_check && !active && (!istype(worn_glasses) || !(worn_glasses.obj_flags & NIF_HUD_GRANTER)))
+		linked_mob.balloon_alert("no compatible eyewear!")
+		return FALSE
+
+	. = ..() // active = !active
+	if(!.)
+		return FALSE
+
+	if(!active)
+		if(hud_type)
+			var/datum/atom_hud/hud = GLOB.huds[hud_type]
+			hud.hide_from(linked_mob)
+		if(hud_trait)
+			REMOVE_TRAIT(linked_mob, hud_trait, NIFSOFT_TRAIT)
+
+		for(var/trait in added_eyewear_traits)
+			REMOVE_TRAIT(linked_mob, trait, NIFSOFT_TRAIT)
+
+		if(eyewear_check)
+			if(!istype(worn_glasses)) // Really non-ideal situation, but it's better than a runtime.
+				return FALSE
+
+			UnregisterSignal(worn_glasses, COMSIG_ITEM_PRE_UNEQUIP)
+
+		return TRUE
```

Comment:
could you split this and the below section into separate procs and call those please